src/back/spv/writer.rs

```diff
@@ -1114,80 +1129,83 @@ impl Writer {
-                                    left_result_type_id,
-                                    id,
-                                    left_id,
-                                    right_id,
-                                ),
-                                left_lookup_ty,
+                            left_lookup_ty,
+                        ),
+                        (Dimension::Matrix, Dimension::Scalar { .. }) => (
+                            super::instructions::instruction_matrix_times_scalar(
+                                left_result_type_id,
+                                id,
+                                left_id,
+                                right_id,
                             ),
-                            crate::TypeInner::Vector { .. } => (
-                                super::instructions::instruction_matrix_times_vector(
-                                    right_result_type_id,
-                                    id,
-                                    left_id,
-                                    right_id,
-                                ),
-                                right_lookup_ty,
+                            left_lookup_ty,
+                        ),
+                        (Dimension::Matrix, Dimension::Vector) => (
+                            super::instructions::instruction_matrix_times_vector(
+                                right_result_type_id,
+                                id,
+                                left_id,
+                                right_id,
                             ),
-                            crate::TypeInner::Matrix { .. } => (
-                                super::instructions::instruction_matrix_times_matrix(
-                                    left_result_type_id,
-                                    id,
-                                    left_id,
-                                    right_id,
-                                ),
-                                left_lookup_ty,
+                            right_lookup_ty,
+                        ),
+                        (Dimension::Matrix, Dimension::Matrix) => (
+                            super::instructions::instruction_matrix_times_matrix(
+                                left_result_type_id,
+                                id,
+                                left_id,
+                                right_id,
                             ),
-                            _ => unreachable!(),
-                        },
-                        crate::TypeInner::Scalar { kind, .. } => match kind {
-                            crate::ScalarKind::Sint | crate::ScalarKind::Uint => (
+                            left_lookup_ty,
+                        ),
+                        (Dimension::Scalar, Dimension::Scalar)
+                            if left_ty_inner.scalar_kind() == Some(crate::ScalarKind::Sint)
```

Comment:
let's first put the one with `Float` condition, and this arm would go after it. It doesn't need the `if` at all, since naturally it's either integer or unreachable. We can safely assume it's integer.

Reply:
True yes, I used this order as I followed the ScalarKind enum order. Makes sense to put Float first.